listings/applications/components_dependencies/src/countable_dep_switch.cairo

```diff
@@ -25,3 +25,73 @@ pub mod countable_component {
     }
 // ANCHOR_END: impl
 }
+
+#[starknet::contract]
+mod MockContract {
+    use super::countable_component;
+    use components::switchable::ISwitchable;  
+
+    component!(path: countable_component, storage: counter, event: CountableEvent);
+
+     #[storage]
+    struct Storage {
+        #[substorage(v0)]
+        counter: countable_component::Storage,
+        switch:bool,
+    }
+
+    #[event]
+    #[derive(Drop, starknet::Event)]
+    enum Event {
+        CountableEvent: countable_component::Event,
+    }
+
+    #[abi(embed_v0)]
+    impl CountableImpl = countable_component::Countable<ContractState>;
+    #[abi(embed_v0)]
+    impl Switchable of ISwitchable<ContractState> {
+        fn switch(ref self: ContractState) {
+            self.switch.write(!self.switch.read());
+        }
+
+        fn is_on(self: @ContractState) -> bool {
+            self.switch.read()
+        }
+    }
+
+    #[constructor]
+    fn constructor(ref self: ContractState) {
+        self.switch.write(true);
+    }
```

Comment:
For the mock contract, you can remove constructor, remove the implementation of `switch` and make `is_on` always return `true`.